lightning/src/ln/onion_utils.rs

```diff
@@ -991,6 +991,29 @@ where
 		_ => unreachable!(),
 	};
 
+	// Check that there is at least enough data for an hmac, otherwise none of the checking that we may do makes sense.
+	// Also prevent slice out of bounds further down.
+	if encrypted_packet.data.len() < 32 {
+		log_warn!(
+			logger,
+			"Non-attributable failure encountered on route {}",
+			path.hops.iter().map(|h| h.pubkey.to_string()).collect::<Vec<_>>().join("->")
+		);
```

Comment:
Is it possible for someone to send us a bunch of HTLC fails and DoS our logs? May want to make it `trace` if so

Reply:
Yes, the intention here is to provide visibility into non-attributable failures if they'd happen. DoS logs is not a thing on a proper deployment with log rotation, is it? ;)